app/src/main/java/ch/epfl/qedit/backend/database/Util.java

```diff
@@ -195,8 +219,17 @@ public static void extractStringPool(
                 return;
             }
         }
+        StringPool stringPool = new StringPool(result);
+        stringPool.setLanguageCode(language);
+        future.complete(stringPool);
+    }
 
-        future.complete(new StringPool(result));
+    public static void extractTreasureHunt(
+            CompletableFuture<Boolean> future, DocumentSnapshot doc) {
+        Boolean treasureHunt = doc.getBoolean("treasureHunt");
+        if (treasureHunt == null)
+            formatError(future, "A quiz must specified if it is a treasure hunt or not");
```

Comment:
small typo here, it should by spelled 'specify' ;).

Reply:
thanks'